runtime/lua/vim/lsp.lua

```diff
@@ -1313,5 +1313,29 @@ end
 -- Define the LspDiagnostics signs if they're not defined already.
 require('vim.lsp.diagnostic')._define_default_signs_and_highlights()
 
+
+--- Registry for client side code action commands.
+---
+--- This is an extension point for plugins to handle custom commands which are
+--- not part of the core language server protocol specification.
+---
+--- The codeAction response contains commands which are either executed with a
+--- function from this registry, or via `vim.lsp.buf.execute_command` if there is
+--- no entry for a given command.
+---
+--- This is a table where the key must be a command name and the value must be
+--- function with two arguments.
+---
+--- The first argument to the function will be the `Command`:
+--    Command
+--      title: String
+--      command: String
+--      arguments?: any[]
+--
+--- The second argument are the params which were sent as payload with the
+--- original codeAction request.
+lsp.commands = {}
```

Comment:
I'm thinking that I'd like to add things to `commands` via a function that can validate the function/object/whatever at registration time (not 100% sure though).

Another thing  to consider would be that some commands may want to be registered directly on the client, so we might want to provide a way to do that -- similar to how we do that with handlers.

Reply:
> I'm thinking that I'd like to add things to commands via a function that can validate the function/object/whatever at registration time (not 100% sure though).


Added some basic validation logic
People could still override the commands completely, not sure if we want to prevent that as well?
`handlers` are currently entirely open too.

> Another thing to consider would be that some commands may want to be registered directly on the client, so we might want to provide a way to do that -- similar to how we do that with handlers.

Would you be okay to follow up on that once somebody asks for it? I think it would be a non-breaking extension